### PR TITLE
[CARBONDATA-1903] Fix code issues in carbondata

### PR DESCRIPTION
**Pull request by xuchuanyin**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
    `No`
 - [x] Any backward compatibility impacted?
    `No`
 - [x] Document update required?
    `No`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        `No, only fixed code related issues.`
        - How it is tested? Please attach test report.
        `Tested in local machine`
        - Is it a performance related change? Please attach the performance test report.
        `No, only fixed code related issues.`
        - Any additional information to help reviewers in testing this change.
        `No`
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
        `Not related`

Modification
===

1. Remove unused code like `FileUtil`

2. Fix/Optimize code issues in carbondata

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2126/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/902/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2412/

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java [hunk 1]
```diff
@@ -134,9 +134,8 @@ public BlockletDataMap get(TableBlockIndexUniqueIdentifier identifier)
   @Override
   public BlockletDataMap getIfPresent(
       TableBlockIndexUniqueIdentifier tableSegmentUniqueIdentifier) {
-    BlockletDataMap dataMap = (BlockletDataMap) lruCache
+    return (BlockletDataMap) lruCache
         .get(tableSegmentUniqueIdentifier.getUniqueTableSegmentIdentifier());
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: move .get to previous line

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataRefNodeWrapper.java [hunk 2]
```diff
@@ -124,11 +124,10 @@ public DimensionRawColumnChunk getDimensionChunk(FileHolder fileReader, int bloc
   private DimensionColumnChunkReader getDimensionColumnChunkReader() throws IOException {
     ColumnarFormatVersion version =
         ColumnarFormatVersion.valueOf(blockInfos.get(index).getDetailInfo().getVersionNumber());
-    DimensionColumnChunkReader dimensionColumnChunkReader = CarbonDataReaderFactory.getInstance()
+    return CarbonDataReaderFactory.getInstance()
         .getDimensionColumnChunkReader(version,
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataRefNodeWrapper.java]: move .getDimensionColumnChunkReader to previous line

File: hadoop/src/main/java/org/apache/carbondata/hadoop/util/SchemaReader.java [hunk 3]
```diff
@@ -75,9 +75,8 @@ public static TableInfo getTableInfo(AbsoluteTableIdentifier absoluteTableIdenti
         new ThriftWrapperSchemaConverterImpl();
     CarbonTableIdentifier carbonTableIdentifier =
         absoluteTableIdentifier.getCarbonTableIdentifier();
-    TableInfo tableInfo = thriftWrapperSchemaConverter
+    return thriftWrapperSchemaConverter
```
Comment (jackylk) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/util/SchemaReader.java]: change to 
```
      foo(
       a,
       b,
       ...)
```

File: processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java [hunk 4]
```diff
@@ -298,8 +298,7 @@ private void readAndLoadDataFromSortTempFiles() throws Exception {
         try {
           dataHandler.closeHandler();
         } catch (CarbonDataWriterException e) {
-          LOGGER.error(e);
-          throw new Exception("Problem loading data during compaction: " + e.getMessage());
+          LOGGER.error(e, "Error in close data handler");
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java]: not throwing exception to caller?

Comment (xuchuanyin) [reply processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java]: It‘s better not to throw Exception in finally block, so we just swallow this exception.

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2437/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/924/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2153/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2162/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/933/

File: processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java (hunk 4 above)
Comment (jackylk) [reply processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java]: I think it is better not to hide the exception even in finally block, there are multiple operations inside dataHandler.closeHandler(), let it throw exception to caller then we can know which operation cause the exception.
In line 292, 295, 301, please modify to use `new Exception(msg, e) to include the original exception in the new exception.

**Comment by xuchuanyin** [comment]
retest this please

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2241/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1017/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2499/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2242/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1018/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2246/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1023/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2505/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2251/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1028/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2507/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2514/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2262/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1040/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2285/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1069/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2291/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1075/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1078/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2307/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1091/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1110/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2330/

File: processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java (hunk 4 above)
Comment (xuchuanyin) [reply processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java]: OK~

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2352/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1135/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2355/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1137/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1139/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2568/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1143/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2373/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2404/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1181/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2435/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1211/

**Comment by QiangCai** [comment]
retest this please

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2511/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1288/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2673/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2518/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1299/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by QiangCai** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2531/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1305/

**Pull request closed without merging**